server-mock/server/get-minipool-available.go

```diff
@@ -4,12 +4,32 @@ import (
 	"net/http"
 
 	apiv2 "github.com/nodeset-org/nodeset-client-go/api-v2"
+	"github.com/nodeset-org/nodeset-client-go/server-mock/db"
 )
 
 func (s *NodeSetMockServer) getMinipoolAvailable(w http.ResponseWriter, r *http.Request) {
 	data := apiv2.MinipoolAvailableData{}
-	handleSuccess(w, s.logger, data)
 
-	s.logger.Info("Fetched minipool available")
+	// Get the requesting node
+	session := s.processAuthHeader(w, r)
+	if session == nil {
+		return
+	}
+	node := s.getNodeForSession(w, session)
+	if node == nil {
+		return
+	}
+
+	// Get minipool available count from database
+	db := db.NewDatabase(s.logger)
```

Comment:
This is always going to return 0 since you're making a new DB here, isn't it? I think you want to have the manager get it from its own DB instead of querying the DB directly here.

Reply:
updated